--- fsharp.org/Makefile
+++ fsharp/Makefile
@@ -1,12 +1,6 @@
 include $(topsrcdir)mono/config.make
 
-.PHONY: restore build build-proto
-
-restore:
-	MONO_ENV_OPTIONS=$(monoopts) mono .nuget/NuGet.exe restore packages.config -PackagesDirectory packages -ConfigFile ./NuGet.Config
-	chmod u+x packages/FSharp.Compiler.Tools.4.1.27/tools/fsi.exe 
-	chmod u+x packages/FsLexYacc.7.0.6/build/fslex.exe
-	chmod u+x packages/FsLexYacc.7.0.6/build/fsyacc.exe
+.PHONY: build build-proto
 
 # Make the proto using the bootstrap, then make the final compiler using the proto
 # We call MAKE sequentially because we don't want build-final to explicitly depend on build-proto,
@@ -20,7 +14,6 @@ all:
 	@echo monolibdir=$(monolibdir)
 	@echo monobindir=$(monobindir)
 	@echo -----------
-	$(MAKE) restore
 	$(MAKE) build-proto
 	$(MAKE) build
 
